AutoRouteManager.php

```diff
@@ -98,7 +98,13 @@ private function getUriContextsForDocument(UriContextCollection $uriContextColle
 
         foreach ($locales as $locale) {
             if (null !== $locale) {
-                $this->adapter->translateObject($uriContextCollection->getSubjectObject(), $locale);
+                $subjectObject = $this->adapter->translateObject($uriContextCollection->getSubjectObject(), $locale);
+
+                if (null !== $subjectObject) {
+                    $uriContextCollection->setSubjectObject($subjectObject);
+                } else {
+                    @trigger_error('AdapterInterface::translateObject() has to return the subjectObject as of version 1.1.', E_USER_DEPRECATED);
```

Comment:
this is a BC hack so that the system does not break if the adapter does
not return the translated object.

lets create an issue with things to remove / rename in 2.0